Fj 2024 lesson 4

vladimir-arsenov opened the pull request.

vladimir-arsenov committed:
Task 3 is completed

lParzivall commented on src/main/kotlin/Main.kt:
```diff
@@ -0,0 +1,158 @@
+import io.ktor.client.*
```
лучше импортить только то, что тебе нужно

lParzivall replied:
не хватает разделения по пакетам client service dto

lParzivall commented on src/main/kotlin/Main.kt:
```diff
@@ -0,0 +1,158 @@
+}
+
+// Задание #2
+suspend fun getNews(count: Int = 100): List<News> {
```
вынести в класс

lParzivall commented on src/main/kotlin/Main.kt:
```diff
@@ -0,0 +1,158 @@
+
+
+// Задание #3
+fun List<News>.getMostRatedNews(count: Int, period: ClosedRange<LocalDate>): List<News> {
```
вынести в класс

lParzivall commented on src/main/kotlin/Main.kt:
```diff
@@ -0,0 +1,158 @@
+//}
+
+// Задание #4
+fun saveNews(path: String, news: Collection<News>) {
```
вынести в класс

lParzivall commented on src/main/kotlin/Main.kt:
```diff
@@ -0,0 +1,158 @@
+    var page = 0
+
+    try {
+        while (page * count.coerceAtMost(100) < count) {
```
сомнительно, но ОКЭЙ

vladimir-arsenov replied:
А как стоило сделать?

lParzivall commented on src/main/kotlin/DSL.kt:
```diff
@@ -0,0 +1,98 @@
+fun main() {
+    val newsList = yamlNewsList {
```
круто

lParzivall replied:
еще бы пример файликов закомиченных увидеть, но не оябзательно

vladimir-arsenov committed:
Organized project with dto, client, service, dsl packages
Moved methods from Main.kt to new classes: NewsService.kt and NewsClient.kt
Added news.csv and newsPrettyPrintExample.yaml
